x-pack/plugin/sql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/sql/qa/mixed_node/SqlCompatIT.java

```diff
@@ -135,4 +138,20 @@ private void testNullsOrderWithMissingOrderSupport(RestClient client) throws IOE
         return rows.stream().map(row -> (Integer) row.get(0)).collect(Collectors.toList());
     }
 
+    public static String sqlQueryEntityWithOptionalMode(String query, Version bwcVersion) throws IOException {
+        XContentBuilder json = XContentFactory.jsonBuilder().startObject();
+        json.field("query", query);
+        if (bwcVersion.before(Version.V_7_12_0)) {
+            // a bug previous to 7.12 caused a NullPointerException when accessing displaySize in ColumnInfo. The bug has been addressed in
+            // https://github.com/elastic/elasticsearch/pull/68802/files
+            // #diff-2faa4e2df98a4636300a19d9d890a1bd7174e9b20dd3a8589d2c78a3d9e5cbc0L110
```

Comment:
Add one more comment stating that the workaround for versions between 7.10.3 and 7.12.0 is to use the `driver` mode.

Reply:
👍